### PR TITLE
Modify duplicate detection for Applicants and Positions

### LINKED ISSUE #161: [PE-D] Flaws in duplicate checking of applicants
### To reproduce
1. Any valid add-applicant cmd with `n/sally`
2. Any valid add-applicant cmd with `n/Sally`

### Expected
Given that the names are the same except for case, I would expect them to be seen as the same person.

### Actual
They are distinct persons. This affects user friendliness of the app as we cannot expect the user to be case-correct all the time.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: sunjc826/ped#7

### PR DESCRIPTION
**Pull request by LMAOboxhack**
Closes #161.

Updated duplicate detection of Applicants and Positions, by overriding the `equals` method for `Name` and `Title` respectively to do a case-insensitive comparison.

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddApplicantCommand.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
     public static final String MESSAGE_SUCCESS = "New applicant added: %1$s";
     public static final String MESSAGE_DUPLICATE_APPLICANT = "This applicant already exists in MrTechRecruiter";
```
Comment (BenjaminTan99) [inline src/main/java/seedu/address/logic/commands/AddApplicantCommand.java:37]:
```suggestion
    public static final String MESSAGE_DUPLICATE_APPLICANT = "This applicant already exists in MrTechRecruiter.";
```
For consistency

File: src/main/java/seedu/address/logic/commands/AddApplicantCommand.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
 
     public static final String MESSAGE_SUCCESS = "New applicant added: %1$s";
```
Comment (BenjaminTan99) [inline src/main/java/seedu/address/logic/commands/AddApplicantCommand.java:36]:
```suggestion
    public static final String MESSAGE_SUCCESS = "New applicant added: %1$s.";
```
For consistency

File: src/main/java/seedu/address/logic/descriptors/InvalidFilterApplicantDescriptor.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
+package seedu.address.logic.descriptors;
+
+import static seedu.address.model.applicant.ApplicantMatchesFiltersPredicate.passesFilter;
+
+import seedu.address.model.Model;
+
+/**
+ * A descriptor used to verify the validity of a FilterApplicantDescriptor.
+ */
+public class InvalidFilterApplicantDescriptor extends FilterApplicantDescriptor {
```
Comment (BenjaminTan99) [inline src/main/java/seedu/address/logic/descriptors/InvalidFilterApplicantDescriptor.java:10]:
```suggestion
public class FilterApplicantDescriptorVerifier extends FilterApplicantDescriptor {
```
Invalid sounds a little strange, but I get where you're coming from.

**Review by BenjaminTan99 — COMMENTED** [review]
Some minor changes, LGTM otherwise.

File: src/main/java/seedu/address/logic/commands/AddApplicantCommand.java (hunk 1 above)
Comment (LMAOboxhack) [reply src/main/java/seedu/address/logic/commands/AddApplicantCommand.java:37]: Thanks! Fixed.

File: src/main/java/seedu/address/logic/commands/AddApplicantCommand.java (hunk 2 above)
Comment (LMAOboxhack) [reply src/main/java/seedu/address/logic/commands/AddApplicantCommand.java:36]: Thanks! Fixed.

File: src/main/java/seedu/address/logic/descriptors/InvalidFilterApplicantDescriptor.java (hunk 3 above)
Comment (LMAOboxhack) [reply src/main/java/seedu/address/logic/descriptors/InvalidFilterApplicantDescriptor.java:10]: Thanks! That does sound better.

**Review by Harry-Gao-H — APPROVED** [review]
LGTM

**Review by BenjaminTan99 — APPROVED** [review]
LGTM

**Pull request merged**